### PR TITLE
[WIP - ILLUSTRATION_ONLY] Switch to custom test container

### PR DESCRIPTION
**Pull request by onobc**
@dmfrey

### CODE REVIEW

File: functions/common/function-test-support/pom.xml [hunk 1]
```diff
@@ -47,8 +47,12 @@
         <dependency>
             <groupId>org.testcontainers</groupId>
             <artifactId>testcontainers</artifactId>
-            <version>${test-containers.version}</version>
+            <scope>compile</scope>
```
Comment (artembilan) [inline functions/common/function-test-support/pom.xml:50]: But you said yourself in other place that this one is a default scope…

File: functions/common/function-test-support/src/main/java/org/springframework/cloud/fn/test/support/xmpp/XmppContainer.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+	public XmppContainer() {
+		this(DEFAULT_IMAGE_NAME);
+	}
+
+	public XmppContainer(DockerImageName dockerImageName) {
+		super(dockerImageName);
+		dockerImageName.assertCompatibleWith(DEFAULT_IMAGE_NAME);
+		withExposedPorts(XMPP_INTERNAL_PORT);
+		withClasspathResourceMapping("xmpp/conf", "/var/lib/openfire/conf", BindMode.READ_ONLY);
+		withCommand("-demoboot");
```
Comment (artembilan) [inline functions/common/function-test-support/src/main/java/org/springframework/cloud/fn/test/support/xmpp/XmppContainer.java:39]: Doesn’t look like any  of these give us a justification of a custom impl. Why all of these cannot go directly to that interface you have introduced to start a generic container in tests?

File: functions/consumer/xmpp-consumer/src/test/java/org/springframework/cloud/fn/consumer/xmpp/XmppConsumerConfigurationTests.java [hunk 3]
```diff
@@ -55,30 +54,19 @@
...
+public class XmppConsumerConfigurationTests implements XmppTestContainerSupport {
+
+	@DynamicPropertySource
+	static void registerConfigurationProperties(DynamicPropertyRegistry registry) {
+		registry.add("xmpp.factory.user", () -> JOHN_USER);
+		registry.add("xmpp.factory.password", () -> USER_PW);
+		registry.add("xmpp.factory.host", () -> XmppTestContainerSupport.getXmppHost());
+		registry.add("xmpp.factory.port", () -> XmppTestContainerSupport.getXmppMappedPort());
+		registry.add("xmpp.factory.service-name", () -> SERVICE_NAME);
+		registry.add("xmpp.factory.security-mode", () -> "disabled");
```
Comment (artembilan) [inline functions/consumer/xmpp-consumer/src/test/java/org/springframework/cloud/fn/consumer/xmpp/XmppConsumerConfigurationTests.java:68]: Looks like some of these are really not dynamic. What is wrong with specifying via removed `properties`?

File: functions/common/function-test-support/pom.xml (hunk 1 above)
Comment (onobc) [reply functions/common/function-test-support/pom.xml:50]: I did not update the original changes - it was copied over.

File: functions/consumer/xmpp-consumer/src/test/java/org/springframework/cloud/fn/consumer/xmpp/XmppConsumerConfigurationTests.java (hunk 3 above)
Comment (onobc) [reply functions/consumer/xmpp-consumer/src/test/java/org/springframework/cloud/fn/consumer/xmpp/XmppConsumerConfigurationTests.java:68]: Agreed, I had a similar question in original PR - this was just copy over from the PR.

File: functions/common/function-test-support/src/main/java/org/springframework/cloud/fn/test/support/xmpp/XmppContainer.java (hunk 2 above)
Comment (onobc) [reply functions/common/function-test-support/src/main/java/org/springframework/cloud/fn/test/support/xmpp/XmppContainer.java:39]: Yeh, good point. As long as we can access these methods, then that would simplify things even more.

**Comment by onobc** [comment]
Superceded by https://github.com/spring-cloud/stream-applications/commit/9a6f05def7f69046cd8c462db480464960559163

**Pull request closed without merging**